src/App.js

```diff
@@ -1,24 +1,73 @@
-import logo from './logo.svg';
-import './App.css';
+import { useEffect, useState } from 'react'
+import BarList from './components/BarList'
+import './App.css'
+import axios from 'axios'
+import UserInput from './components/UserInput'
 
 function App() {
+  const [barList, setBarList] = useState([])
+  const [currentLocation, setCurrentLocation] = useState({})
+  const [locationUnavailable, setLocationUnavailable] = useState(true)
+
+  const baseURL = new URL('https://api.openbrewerydb.org/v1/breweries')
+  
+  async function fetchData() {
+    if ("geolocation" in navigator) {
+      const location = await getLocation()
+      await findBarsHandler(location)
+    } else {
+      await findBarsHandler()
+    }
+  }
+
+  async function getLocation() {
+    const promise = new Promise((resolve, reject) => {
+      navigator.geolocation.getCurrentPosition(resolve, reject)
+    })
+    try {
+      const {coords: location} = await promise
+      setCurrentLocation(location)
+      setLocationUnavailable(false)
+      return location
+    } catch (error) {
+      console.log(error)
+    }
+  }
+  
+  async function findBarsHandler(location=currentLocation) {
+    try {
+      if (!!location.latitude) {
+        const query = `${location.latitude},${location.longitude}`
+        baseURL.searchParams.append("by_dist", query)
+      } else if (location.length === 5) {
```

Comment:
Instead of overriding your location data with a string representing the postal code, you should probably append it as a property on the location object. `location.postal` or something so that your previous check for a latitude doesn't throw an error when it looks for `.latitude` on a number or undefined object.